python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -357,13 +357,6 @@ def update(self):
         except Exception as e:
             self.prom_metrics.update_loop_exceptions.inc()
             logger.exception("StandardAutoscaler: Error during autoscaling.")
-            # Don't abort the autoscaler if the K8s API server is down.
-            # https://github.com/ray-project/ray/issues/12255
-            is_k8s_connection_error = self.config["provider"][
-                "type"
-            ] == "kubernetes" and isinstance(e, MaxRetryError)
-            if not is_k8s_connection_error:
-                self.num_failures += 1
```

Comment:
Context: This code only affects "type" Kubernetes, which is being deleted.

For KubeRay, it's fine for the autoscaler to exit -- K8s will restart it.

Reply:
(Though we could think about whether K8s crashlooping is ok or we want to handle things in a more controlled way.)